cpp/include/cudf_test/testing_main.hpp

```diff
@@ -145,6 +145,32 @@ inline auto parse_cudf_test_opts(int argc, char** argv)
   }
 }
 
+/**
+ * @brief Sets up stream mode memory resource adaptor
+ *
+ * The resource adaptor is only set as the current device resource if the
+ * stream mode is enabled.
+ *
+ * The caller must keep the return object alive for the life of the test runs.
+ *
+ * @param cmd_opts Command line options returned by parse_cudf_test_opts
+ * @return Memory resource adaptor
+ */
+inline auto make_stream_mode_adaptor(cxxopts::ParseResult const& cmd_opts)
```

Comment:
I have a mild preference for actually declaring the return type here instead of using auto.  Gives the caller a little more context with what's going on.

Reply:
Normally I prefer the return types as well. The type here is something created through templated RMM classes only to be passed back to an RMM function. This means RMM could change the type/name on us and would then require this code to be updated even though we don't actually rely on the type for anything--we don't make calls to it, we just pass it to `set_current_device_resource`. The returned object then just needs to stay alive for the life of the tests. Leaving it as `auto` here helps future proofs this code a bit.